alf/layers_test.py

```diff
@@ -151,6 +151,64 @@ def test_parallel_conv_transpose(self,
             y = convt(x)
             self.assertLess((y - py[:, i, :]).abs().max(), 1e-5)
 
+    @parameterized.parameters(
+        ("rbf", 8, 8, 0.1),
+        ("poly", 4, 8, None),
+        ("cheb", 4, 8, None),
+        ("haar", 8, 8, None),
+        ("rbf", 7, 8, 0.1),
+        ("haar", 7, 7, None),
+        ("unimplemented", 3, 8, None),
+    )
+    def test_fixed_decoding_layer(self, basis_type, input_size, output_size,
+                                  sigma):
+        batch_size = 3
+
+        if (basis_type == "rbf" and input_size != output_size) or \
+           (basis_type == "haar" and (input_size & (input_size - 1)) != 0) or \
```

Comment:
for haar, please compare to an explicit matrix to make sure it's correctly constructed.

Reply:
Now added explicit tests against several explicit Haar matrices.